src/components/button-toggle/button-toggle.ts

```diff
@@ -0,0 +1,341 @@
+  host: {
+    '(click)': 'onClick($event)'
+  }
+})
+export class MdButtonToggle implements OnInit {
+  /** Whether or not this button toggle is checked. */
+  private _checked: boolean = false;
+
+  /** Type of the button toggle. Either 'radio' or 'checkbox'. */
+  private _type: ToggleType;
+
+  /** The unique ID for this button toggle. */
+  @HostBinding()
+  @Input()
+  id: string;
+
+  /** HTML's 'name' attribute used to group radios for unique selection. */
+  @Input()
+  name: string;
+
+  /** Whether or not this button toggle is disabled. */
+  private _disabled: boolean = null;
+
+  /** Value assigned to this button toggle. */
+  private _value: any = null;
+
+  /** Whether or not the button toggle is a single selection. */
+  private _isSingleSelector: boolean = null;
+
+  /** The parent button toggle group (exclusive selection). Optional. */
+  buttonToggleGroup: MdButtonToggleGroup;
+
+  /** The parent button toggle group (multiple selection). Optional. */
+  buttonToggleGroupMultiple: MdButtonToggleGroupMultiple;
+
+  /** Event emitted when the group value changes. */
+  @Output()
+  change: EventEmitter<MdButtonToggleChange> = new EventEmitter<MdButtonToggleChange>();
+
+  constructor(@Optional() toggleGroup: MdButtonToggleGroup,
+              @Optional() toggleGroupMultiple: MdButtonToggleGroupMultiple,
+              public buttonToggleDispatcher: MdUniqueSelectionDispatcher) {
+    this.buttonToggleGroup = toggleGroup;
+
+    this.buttonToggleGroupMultiple = toggleGroupMultiple;
+
+    if (this.buttonToggleGroup) {
+      buttonToggleDispatcher.listen((id: string, name: string) => {
+        if (id != this.id && name == this.name) {
+          this.checked = false;
+        }
+      });
+
+      this._type = 'radio';
+      this.name = this.buttonToggleGroup.name;
+      this.isSingleSelector = true;
+    } else {
+      // Even if there is no group at all, treat the button toggle as a checkbox so it can be
+      // toggled on or off.
+      this._type = 'checkbox';
+      this.isSingleSelector = false;
+    }
+  }
+
+  /** @internal */
+  ngOnInit() {
+    if (this.id == null) {
+      this.id = `md-button-toggle-${_uniqueIdCounter++}`;
+    }
+
+    if (this.buttonToggleGroup && this._value == this.buttonToggleGroup.value) {
+      this._checked = true;
+    }
+
+  }
+
+  get inputId(): string {
+    return `${this.id}-input`;
+  }
+
+  @HostBinding('class.md-button-toggle-checked')
+  @Input()
+  get checked(): boolean {
+    return this._checked;
+  }
+
+  set checked(value: boolean) {
+    if (value && this.isSingleSelector) {
+      // Notify all button toggles with the same name (in the same group) to un-check.
+      this.buttonToggleDispatcher.notify(this.id, this.name);
+
+      if (!this._checked) {
+        this._emitChangeEvent();
+      }
+    }
+
+    this._checked = value;
+  }
+
+  get type(): ToggleType {
+    return this._type;
+  }
+
+  set type(value: ToggleType) {
+    this._type = value;
+  }
```

Comment:
Is it really necessary to have a getter and setter here?
You're just setting the value of the private property. It would be the same, to have a normal property here.